third-party/realdds/src/dds-topic-reader-thread.cpp

```diff
@@ -40,25 +42,31 @@ void dds_topic_reader_thread::run( qos const & rqos )
         DDS_THROW( runtime_error, "on-data-available must be provided" );
 
     _reader = DDS_API_CALL( _subscriber->get()->create_datareader( _topic->get(), rqos ) );
+    _stopped = std::make_shared< eprosima::fastdds::dds::GuardCondition >();
     
     _th = std::thread(
-        [this, name = _topic->get()->get_name()]()
+        [this,
```

Comment:
Do we need both `this` and `shared_from_this` for the same lambda?

Reply:
Accessing the callback functions, `on_sample_lost` etc., through the shared_ptr is not allowed as the code is written (they're not public).